test/libraries/SafeMigration.spec.ts

```diff
@@ -1,325 +1,387 @@
 import { expect } from "chai";
 import hre, { ethers, deployments } from "hardhat";
 import {
-    getSafeWithSingleton,
+    getSafe,
     getSafeSingletonAt,
     safeMigrationContract,
-    getSafeSingletonContract,
-    getSafeL2SingletonContract,
+    getSafeL2Singleton,
     getCompatFallbackHandler,
+    getSafeL1Singleton,
+    getAbi,
 } from "../utils/setup";
-import safeDeploymentData from "../json/safeDeployment.json";
+import deploymentData from "../json/safeDeployment.json";
+import fallbackHandlerDeploymentData from "../json/fallbackHandlerDeployment.json";
 
 import { executeContractCallWithSigners } from "../../src/utils/execution";
+import { SafeMigration } from "../../typechain-types";
 
 const FALLBACK_HANDLER_STORAGE_SLOT = "0x6c9a6c4a39284e37ed1cf53d337577d14212a4870fb976a4366c693b939918d5";
 
 const GUARD_STORAGE_SLOT = "0x4a204f620c8c5ccdca3fd54d003badd85ba500436a431f0cbda4f558c93c34c8";
 
-describe("SafeMigration Library", () => {
-    const migratedInterface = new ethers.utils.Interface(["function masterCopy() view returns(address)"]);
+const migrationPaths = [
+    {
+        testSuiteName: "1.3.0 to latest (1.5.0)",
+        from: {
+            safeDeploymentData: { evm: deploymentData.safe130.evm, zksync: deploymentData.safe130.zksync },
+            safeL2DeploymentData: { evm: deploymentData.safe130l2.evm, zksync: deploymentData.safe130l2.zksync },
+        },
+        latest: true,
+    },
+    {
+        testSuiteName: "1.3.0 to 1.4.1",
+        from: {
+            safeDeploymentData: { evm: deploymentData.safe130.evm, zksync: deploymentData.safe130.zksync },
+            safeL2DeploymentData: { evm: deploymentData.safe130l2.evm, zksync: deploymentData.safe130l2.zksync },
+        },
+        // `to` is used when the target contracts are not latest. So, we need to provide the bytecode
+        // of the old version of the contracts
+        to: {
+            safeDeploymentData: { evm: deploymentData.safe141.evm, zksync: deploymentData.safe141.zksync },
+            safeL2DeploymentData: { evm: deploymentData.safe141l2.evm, zksync: deploymentData.safe141l2.zksync },
+            safeCompatFallbackHandler: {
+                evm: fallbackHandlerDeploymentData.compatibilityFallbackHandler141.evm,
+                zksync: fallbackHandlerDeploymentData.compatibilityFallbackHandler141.zksync,
+            },
+        },
+    },
+    {
+        testSuiteName: "1.4.1 to latest (1.5.0)",
```

Comment:
Not highly necessary, but should these tests (migrating to `v1.5.0`) be removed from these?

Reply:
yes, good catch